src/matching/Match_patterns.ml

```diff
@@ -402,11 +444,29 @@ let check2 ~hook mvar_context range_filter (config, equivs) rules
                                 in
                                 Common.push pm matches;
                                 hook pm));
-            super#visit_stmt env x
+            let parent_is_last_stmt_trans =
+              parent_is_last_stmt_trans && self_is_last_stmt
+            in
+            super#visit_stmt
+              (env |> with_imp_ret_ctx ~parent_is_last_stmt_trans)
+              x
           in
-          visit_stmt ()
+          (* Cases like ExprStmt(StmtExpr s) may cause duplicate matches because
+           * it's possible that
+           *   ExprStmt(StmtExpr s)
+           * will match once, and
+           *   s
+           * will also match again. Ideally, the generic ASTs would not include
+           * these cases, but it seems to be the case for at least Julia.
+           *)
+          match x with
+          | { s = ExprStmt ({ e = StmtExpr _; _ }, _); _ } ->
```

Comment:
I think wherever possible, we should allow `Generic_vs_generic` to operate under the simplest assumptions, because those upstream issues are simply the fault of an incorrect translation, and should be changed. This one is on me, because this translation should never have happened in the first place, but I think we should avoid introducing this unrelated code where possible.

Reply:
Originally, I think I had a warning log in there, which somehow got removed. I agree with fixing it earlier. I'm just thinking about catching it for all languages.

Makes me think, maybe it might be a good idea to have a unit test to catch these patterns for any parser introduced/updated.